### ISSUE TITLE
Highlights written into pdf despite setting that should disable this

### ISSUE THREAD

**Issue by alex125** [koreader/koreader#4270]
* KOReader version: latest
* Device: Kobo

#### Issue
Highlights written into pdf despite 'Save document' setting 'Prompt' or 'Disable'
#### Steps to reproduce
Highlight text, immediately after that choose Exit->Power off, reboot and stzart koreader

**Comment by Frenzie** [comment]
Stupid question, but just to make absolutely sure: you mean visible in another program?

**Comment by alex125** [comment]
Yes, the pdf file is saved with the highlight. In koreader, the passage remains gray after it is unhighlighted.  When I open the pdf with Xreader under Linux the passage is black.

**Comment by Frenzie** [comment]
I'll have to try it on my H2O another day. I get a prompt and it properly doesn't save the highlight in the emulator with regular Exit → Exit. (Power off is only available on devices but in principle I think it's supposed to be pretty much the same as exit just powering off after.)

**Comment by Frenzie** [comment]
And when you say latest, you mean `v2015.11-1868-gb8d9598_2018-10-06`?

**Comment by alex125** [comment]
v2015.11-1877-g76a3d20_2018-10-11 but the bug was also there in previous versions, it's just that I couldn't reproduce it and only now found out that it only happens with Exit->Power off. Exit->Exit works ok.

**Comment by Frenzie** [comment]
I'm not really sure at a quick look where it might be coming from. `Device:SaveSettings` doesn't do anything on Kobo or possibly anywhere. So then there's the close event, but that's overall more likely to result in unsaved settings than erroneously saved settings.

https://github.com/koreader/koreader/blob/fa0117bb2a02d0984bded06144c93218a85b21cb/frontend/ui/uimanager.lua#L55-L65

I should check first if I can reproduce I suppose, but I could quickly take a peek at some of the code involved.

**Comment by alex125** [comment]
It also happens with Exit->Sleep followed by Exit->Exit (but only when WiFi is on). To reproduce: 1. turn on Wifi, 2. highlight a word, 3. Exit->Sleep and turn on again (or close and open sleep cover), 4. Exit-> Exit, 5. Start koreader -> pdf file is changed.

**Comment by Frenzie** [comment]
Relevant code paths:

https://github.com/koreader/koreader/blob/717184f1d65a147d90e12ab93afcad2b48d0e310/frontend/apps/reader/readerui.lua#L584-L610

https://github.com/koreader/koreader/blob/717184f1d65a147d90e12ab93afcad2b48d0e310/frontend/document/document.lua#L102-L111

https://github.com/koreader/koreader/blob/717184f1d65a147d90e12ab93afcad2b48d0e310/frontend/document/pdfdocument.lua#L181-L186

Now why could that sometimes be true even when discarding the changes…
